[backport to 3.6] bpo-28929: Link the documentation to its source file on GitHub

Mariatta opened the pull request:
Change the documentation's `Show Source` link on the left menu
to GitHub source file.

(cherry picked from commit 23bafa294c75c20cb85ae5d97d7571a3a0ad8dd3)

brettcannon commented:
Thanks for the PR, but https://github.com/python/cpython is a cpython mirror repository; pull requests are not accepted on this repo. Please submit bugs and patches at https://bugs.python.org.

Mariatta committed:
bpo-28929: Link the documentation to its source file on GitHub

Change the documentation's `Show Source` link on the left menu
to GitHub source file.

(cherry picked from commit 23bafa294c75c20cb85ae5d97d7571a3a0ad8dd3)

methane commented on Doc/tools/templates/customsourcelink.html:
```diff
@@ -3,8 +3,14 @@
-      <li><a href="{{ pathto('_sources/' + sourcename, true)|e }}"
-            rel="nofollow">{{ _('Show Source') }}</a></li>
+      <li>
+        {% if version == "3.7" %}
```
Should this updated when Python 3.7 is released?
I think 2.7, 3.5, and 3.6 branch should not have this if statement.
Or could we pass variable like `doc_source_url_prefix = "https://github.com/python/cpython/blob/3.6/Doc/"` from conf?

Mariatta replied:
Thanks for the review, @methane :)
I removed the if statement as suggested.
Branches 2.7, 3.5, and 3.6 don't have the if statement anymore.

Mariatta committed:
remove if statement
